x-pack/plugins/maps/public/shared/layers/vector_layer.js

```diff
@@ -364,11 +375,12 @@ export class VectorLayer extends AbstractLayer {
     }
   }
 
-
   _assignIdsToFeatures(featureCollection) {
     for (let i = 0; i < featureCollection.features.length; i++) {
       const feature = featureCollection.features[i];
-      feature.properties[FEATURE_ID_PROPERTY_NAME] = (typeof feature.id === 'string' || typeof feature.id === 'number')  ? feature.id : i;
+      const id = generateNumericalId();
+      feature.properties[FEATURE_ID_PROPERTY_NAME] = id;
+      feature.id = id;
```

Comment:
This could be a breaking change. What if the vector source populates `id` field and users expect to see their `id` value in a tooltip and now they see our implementation value. Maybe not a big deal

Reply:
`Feature#id` is not accessible as a tooltip-value. Only index-pattern properties are accessible, and they are put as-is under `Feature#properties`